src/t_zset.c

```diff
@@ -3082,6 +3082,24 @@ void zscoreCommand(client *c) {
     }
 }
 
+void zmscoreCommand(client *c) {
+    robj *key = c->argv[1];
+    robj *zobj;
+    double score;
+    zobj = lookupKeyRead(c->db,key);
+    if (zobj != NULL && checkType(c,zobj,OBJ_ZSET)) return;
+
+    addReplyArrayLen(c,c->argc - 2);
+    for (int j = 2; j < c->argc; j++) {
+        /* Treat a missing set the same way as an empty set */
+        if (zobj == NULL || zsetScore(zobj,c->argv[j]->ptr,&score) == C_ERR) {
```

Comment:
I think we should be consistent with other sorted set commands, and return a single nil if the key doesn't exist. Otherwise there is no differentiation between all the items are missing and the key is missing.

Reply:
> I amended the PR to make fetching the zmscore for an empty zset return a list of nil values.
> 
> This was done to be consistent with https://redis.io/commands/hmget ...
>
> >> Because non-existing keys are treated as empty hashes, running HMGET against a non-existing key will return a list of nil values.

- `get` returns a string or nil
- `mget` returns a list of (string or nil)
- `hget` returns a string or nil
- `hmget` returns a list of (string or nil)
- `zscore` returns a stringified float or nil
- `zmscore` returns a list of (stringified float or nil) to be consistent